raft/log_unstable.go

```diff
@@ -72,6 +98,42 @@ func (u *unstable) maybeTerm(i uint64) (uint64, bool) {
 	return u.entries[i-u.offset].Term, true
 }
 
+// nextEntries returns the unstable entries that are not already in the process
+// of being written to storage.
+func (u *unstable) nextEntries() []pb.Entry {
```

Comment:
Yes it would, but I wanted to keep the nil return value for cases where there are no entries to return. This is placed directly in a `Ready.Entries`, so it's possible that users of the library (incorrectly) compare the slice against nil to check whether anything needs to be appended. This isn't the PR to change that behavior.